chapters/sparse.rst

```diff
@@ -0,0 +1,287 @@
+such as `Scipy.sparse`.
+In Jina, we support three `backends` to create your sparse matrix/Tensor:
+`Scipy`, `TensorFlow` and `Pytorch`.
+You might noticed that `Scipy.sparse` supports different sparse formats,
+while Jina only supports `COO`, `BSR`, `CSR` and `CSC`.
+
+When creating your own sparse matrix,
+we suggest you use `CSR` as default matrix type.
+
+.. list-table:: Sparse Matrix Formats
+   :widths: 25 25 25 25 25
+   :header-rows: 1
+
+   * - ShortName
+     - FullName
+     - Scipy
+     - TensorFlow
+     - Pytorch
+   * - COO
+     - COOrdinate format
+     - Yes
+     - Yes
+     - Yes
+   * - BSR
+     - Block Sparse Row matrix
+     - Yes
+     - No
+     - No
+   * - CSC
+     - Compressed Sparse Column matrix
+     - Yes
+     - No
+     - No
+   * - CSR
+     - Compressed Sparse Row matrix
+     - Yes
+     - No
+     - No
+   * - DIA
+     - Sparse matrix with DIAgonal storage
+     - Yes
+     - No
+     - No
+   * - DOK
+     - Dictionary Of Keys based sparse matrix.
+     - Yes
+     - No
+     - No
+   * - LIL
+     - Row-based list of lists sparse matrix
+     - Yes
+     - No
+     - No
+
+
+Build your Sparse Pipeline In Action
+--------------------------------------
+
+In this pipeline, we will make use of Jina's ``TFIDFTextEncoder`` together with ``PysparnnIndexer`` for encoding and indexing.
+
+::
+
+    project
+    ├── tfidf_vectorizer.pickle
+    ├── encode.yml
+    ├── index.yml
+    ├── flow_index.yml
+    ├── flow_query.yml
+    ├── __init__.py
+    ├── app.py
+    ├── fit_vectorizer.py
+
+
+
+Step 1. Vectorize your data into sparse vector encoding
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+As was mentioned before, ``TFIDFTextEncoder`` was created based on Scikit-learn,
+before using the Encoder, you need to ``fit`` the vectorizer with your training data.
+In this example, we use a simple corpus containing four sentences of text.
+
+.. highlight:: python
+.. code-block:: python
+
+    # fit_vectorizer.py
+    import pickle
+    from sklearn.feature_extraction.text import TfidfVectorizer
+
+    corpus = [
+        'This is the first document.',
+        'This document is the second document.',
+        'And this is the third one.',
+        'Is this the first document?'
+    ]
+
+    vectorizer = TfidfVectorizer()
+    vectorizer.fit(corpus)
+    # Dump the vectorizer fitted on your training data.
+    pickle.dump(vectorizer, open('./tfidf_vectorizer.pickle', 'wb'))
+
+Step 2. Setup Encoder & Indexer YAML configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. highlight:: yaml
+.. code-block:: yaml
+
+    !TFIDFTextEncoder
+    metas:
+      name: tfidf_encoder
+    with:
+      path_vectorizer: ./tfidf_vectorizer.pickle
+
+For the indexer,
+we will use the ``PysparnnIndexer`` with approximate nearest neighbor for sparse data.
+Since we want to store the indexed result, we combined ``PysparnnIndexer`` and ``BinaryPbIndexer`` together.
+
```

Comment:
Create a second YAML file with the following code snippet. The file should be called ??